apps/functions-subscription/src/adapters/azure/functions/subscriptions.ts

```diff
@@ -63,17 +63,68 @@ const makeHandlerKitHandler: H.Handler<
         return pipe({}, H.successJson, H.withStatusCode(202));
       }
     }),
-    RTE.mapLeft((err) => {
-      if (err instanceof ItemAlreadyExists) {
-        return pipe(err, H.toProblemJson, H.problemJson, H.withStatusCode(409));
-      } else {
-        return pipe(err, H.toProblemJson, H.problemJson);
-      }
-    }),
+    RTE.mapLeft(handleSubscriptionProblemJsonResponse),
     RTE.orElseW(RTE.of),
   );
 });
 
 export const makePostSubscriptionHandler = httpAzureFunction(
   makeHandlerKitHandler,
 );
+
+const makeHandlerKitGetSubscriptionHandler: H.Handler<
+  H.HttpRequest,
+  | H.HttpResponse<SubscriptionAPI>
+  | H.HttpResponse<H.ProblemJson, H.HttpErrorStatusCode>,
+  Pick<SystemEnv, 'getSubscription'>
+> = H.of((req: H.HttpRequest) => {
+  return pipe(
+    RTE.ask<Pick<SystemEnv, 'getSubscription'>>(),
+    RTE.apSW(
```

Comment:
you can extract 

```
const getUserId = (req: H.HttpRequest) =>
    RTE.fromEither(parsePathParameter(NonEmptyString, 'userId')(req));

  const getTrialId = (req: H.HttpRequest) =>
    RTE.fromEither(parsePathParameter(NonEmptyString, 'trialId')(req));
```

Reply:
After some refactoring, I decided not to extract those methods; since every function is in its own file, I don't see advantages to have the method wrapped instead of calling the implementation directly within the handler.

Reply:
it's ok, it just reads better.